python-package/xgboost/spark/core.py

```diff
@@ -0,0 +1,810 @@
+import shutil
+import tempfile
+from typing import Iterator, Tuple
+import numpy as np
+import pandas as pd
+from scipy.special import expit, softmax
+from pyspark.ml import Estimator, Model
+from pyspark.ml.param.shared import HasFeaturesCol, HasLabelCol, HasWeightCol, \
+    HasPredictionCol, HasProbabilityCol, HasRawPredictionCol, HasValidationIndicatorCol
+from pyspark.ml.param import Param, Params, TypeConverters
+from pyspark.ml.util import MLReadable, MLWritable
+from pyspark.sql.functions import col, pandas_udf, countDistinct, struct
+from pyspark.sql.types import ArrayType, FloatType
+from xgboost import XGBClassifier, XGBRegressor
+from xgboost.core import Booster
+import cloudpickle
+import xgboost
+from xgboost.training import train as worker_train
+from .utils import get_logger, _get_max_num_concurrent_tasks
+from .data import prepare_predict_data, prepare_train_val_data, convert_partition_data_to_dmatrix
+from .model import (XgboostReader, XgboostWriter, XgboostModelReader,
+                    XgboostModelWriter, deserialize_xgb_model,
+                    get_xgb_model_creator, serialize_xgb_model)
+from .utils import (_get_default_params_from_func, get_class_name,
+                    HasArbitraryParamsDict, HasBaseMarginCol, RabitContext,
+                    _get_rabit_args, _get_args_from_message_list,
+                    _get_spark_session)
+
+from pyspark.ml.functions import array_to_vector, vector_to_array
+
+# Put pyspark specific params here, they won't be passed to XGBoost.
+# like `validationIndicatorCol`, `baseMarginCol`
+_pyspark_specific_params = [
+    'featuresCol', 'labelCol', 'weightCol', 'rawPredictionCol',
+    'predictionCol', 'probabilityCol', 'validationIndicatorCol'
+                                       'baseMarginCol'
+]
+
+_unsupported_xgb_params = [
+    'gpu_id',  # [ML-12862]
```

Comment:
May I ask what "[ML-12862]" means?

Reply:
sorry that's wrong label, I will remove it.